### PR TITLE
Add support for host restriction in sudoers module

### LINKED ISSUE #5702: sudoers module does not support host restrictions
### Summary

Currently the sudoers module is hardcoded to set the host to the magic value `ALL`:

```python
return "{owner} ALL={runas}{nopasswd} {commands}\n".format(owner=owner, runas=runas_str, nopasswd=nopasswd_str, commands=commands_str)
```

This is a problem for me in two different scenarios I have:

1. Some network booted (diskless) systems with a shared `/etc/sudoers.d` (network) filesystem using the host restrictions to constrain which commands are permitted on which host
2. A security policy that requires specifying the host to mitigate the risk of inappropriate escalation being possible if the file is inadvertently copied to the wrong host (e.g. via poorly aimed `scp`, restoring the wrong host's `/etc/sudoers.d` from backup or copying it inappropriately after mounting the disk on a different host in a DR situation - all 3 of which have happened at some point and I'm sure there's more ways to mess it up too)

If accepted, this is a trivial feature to add - it is essentially a carbon-copy of `runas` just before the `=`.

In comparing the code with the `sudoers` manual, I noticed that like the host list - `runas` supports a list of users but that has not been implemented either - just noting it, I have no use case for needing that feature (or supporting a list for the host).

### Issue Type

Feature Idea

### Component Name

sudoers

### Additional Information

```yaml (paste below)

```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by loz-hurst**
##### SUMMARY

Implementation of my feature request: Fixes #5702 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

sudoers

##### ADDITIONAL INFORMATION

Adds ability to set host-based restrictions on privilege escalation managed via sudoers modules. Default behaviour (via default value to new parameter) maintains current behaviour of using the magic `ALL` value (i.e. no restriction).

### CODE REVIEW

File: plugins/modules/sudoers.py [hunk 1]
```diff
@@ -43,6 +43,11 @@
       - Whether a password will be required to run the sudo'd command.
     default: true
     type: bool
+  host:
+    description:
+      - Specify the host the rule is for.
+    default: ALL
+    type: str
```
Comment (felixfontein) [inline plugins/modules/sudoers.py:50]:
```suggestion
    type: str
    version_added: 6.2.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

**Comment by JonEllis** [comment]
Thanks @loz-hurst, I've not used the host option myself, but as the value defaults to ALL as it did before, it seems a safe addition.

**Comment by loz-hurst** [comment]
> Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

Done and thanks for the version added, I wasn't sure what to use there so left it out.

**Comment by russoz** [comment]
> > Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.
> 
> Done and thanks for the version added, I wasn't sure what to use there so left it out.

Hi @loz-hurst Maybe you are missing one git push: The changelog fragment is not in the PR yet.

It would be nice to have one or two tests to demonstrate the use of the new parameter, but the change is very straightforward, so not having new tests will not stop the PR from being merged.

**Comment by loz-hurst** [comment]
> 



> Hi @loz-hurst Maybe you are missing one git push: The changelog fragment is not in the PR yet.

My bad, committed that one to main not patch-1 on my fork.  Should be in there now.

> It would be nice to have one or two tests to demonstrate the use of the new parameter, but the change is very straightforward, so not having new tests will not stop the PR from being merged.

I'll will try and do this but pushed for time...

**Review by russoz — APPROVED** [review]
LGTM

**Comment by loz-hurst** [comment]
> It would be nice to have one or two tests to demonstrate the use of the new parameter, but the change is very straightforward, so not having new tests will not stop the PR from being merged.

I can't see any existing tests for the sudoers module - can someone more familiar with this codebase confirm that before I go ahead and create a new test_sudoers unit?

**Comment by felixfontein** [comment]
There are tests, see tests/integration/targets/sudoers/tasks/main.yml.

**Comment by loz-hurst** [comment]
> There are tests, see tests/integration/targets/sudoers/tasks/main.yml.

Sorry about that, this is my first foray into contributing to anything Ansible related.

I've added a test for this functionality to that suite, CI seems happy (which I presume has run the test, not sure which tasks do the integration tests?).

**Comment by felixfontein** [comment]
Thanks for adding tests! You can see which group this test runs in in https://github.com/ansible-collections/community.general/blob/cd0bb5922eb3f28dd1d660c13b527d61a1df5c3a/tests/integration/targets/sudoers/aliases#L5 - so it's group `2`. You can see your task in one of the runs here: https://dev.azure.com/ansible/community.general/_build/results?buildId=63547&view=logs&j=40c88692-9fb4-5c35-6b69-5e083fc3f5c6&t=899bb9b2-728c-552d-02b1-ac8a85b01cba&l=382

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@loz-hurst thanks for your contribution!
@JonEllis @russoz thanks for reviewing!

**Comment by loz-hurst** [comment]
> Thanks for adding tests! You can see which group this test runs in in
> 
> https://github.com/ansible-collections/community.general/blob/cd0bb5922eb3f28dd1d660c13b527d61a1df5c3a/tests/integration/targets/sudoers/aliases#L5
> 
> - so it's group `2`. You can see your task in one of the runs here: https://dev.azure.com/ansible/community.general/_build/results?buildId=63547&view=logs&j=40c88692-9fb4-5c35-6b69-5e083fc3f5c6&t=899bb9b2-728c-552d-02b1-ac8a85b01cba&l=382

Thank you, I'm learning so much about `community.general` development from this simple change! Really appreciate you taking the time to guide me.

**Pull request merged**